### PR TITLE
support direct connect functionality for load balanced session redirection

### PR DESCRIPTION
**Pull request by jlipps**
## Proposed changes

Bringing the functionality of https://github.com/appium/python-client#direct-connect-urls to wdio.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

- [X] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works - adding an e2e test would be difficult since we'd have to rely on a cloud provider. I did test this with HeadSpin, which supports this cap, and it worked as expected.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by jlipps** [comment]
By the way, I was unable to commit and push this change with git hook verification enabled because of many lint errors unrelated to my change.

**Comment by jlipps** [comment]
yes, can you point me to the right place for that?

**Comment by christian-bromann** [comment]
Give the location of the code you should add a test for it here: https://github.com/webdriverio/webdriverio/blob/master/packages/webdriver/tests/utils.test.js

File: packages/webdriver/src/utils.js [hunk 1]
```diff
@@ -415,3 +415,25 @@ export function getEnvironmentVars({ isW3C, isMobile, isIOS, isAndroid, isChrome
...
 }
+
+/**
+ * Decorate the params object with host updates based on the presence of
+ * directConnect capabilities in the new session response. Note that this
+ * mutates the object.
+ * @param  {Object} params    post-new-session params used to build driver
+ */
+export function setupDirectConnect(params) {
+    log.info(JSON.stringify(params))
```
Comment (christian-bromann) [inline packages/webdriver/src/utils.js]: Do we need this info, if we have another one a couple lines down?

File: packages/webdriver/src/utils.js [hunk 2]
```diff
@@ -415,3 +415,25 @@ export function getEnvironmentVars({ isW3C, isMobile, isIOS, isAndroid, isChrome
...
+    const { directConnectProtocol, directConnectHost, directConnectPort,
+        directConnectPath } = params.capabilities
+    if (directConnectProtocol && directConnectHost && directConnectPort &&
```
Comment (mgrybyk) [inline packages/webdriver/src/utils.js]: can one of these, ex `directConnectPath`, be an empty string?

File: packages/webdriver/src/utils.js (hunk 1 above)
Comment (jlipps) [reply packages/webdriver/src/utils.js]: nope, this was for debugging. will remove

**Comment by jlipps** [comment]
ok, unit tests added.

File: packages/webdriver/src/utils.js (hunk 2 above)
Comment (jlipps) [reply packages/webdriver/src/utils.js]: good point, empty string is probably a valid case.

**Pull request merged**